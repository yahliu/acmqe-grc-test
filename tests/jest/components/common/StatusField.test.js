/*******************************************************************************
 * Licensed Materials - Property of IBM
 * (c) Copyright IBM Corporation 2019. All Rights Reserved.
 *
 * Note to U.S. Government Users Restricted Rights:
 * Use, duplication or disclosure restricted by GSA ADP Schedule
 * Contract with IBM Corp.
 *******************************************************************************/
/* Copyright (c) 2020 Red Hat, Inc. */
/* Copyright Contributors to the Open Cluster Management project */

'use strict'

import React from 'react'
import StatusField from '../../../../src-web/components/common/StatusField'
import renderer from 'react-test-renderer'

describe('StatusField', () => {
  it('renders ok as expected', () => {
    const props = {
      status: 'ok',
      text: 'TextStringOK'
    }
    const component = renderer.create(
      <StatusField  {...props} />
    )
    expect(component.toJSON()).toMatchSnapshot()
  })
  it('renders warning as expected', () => {
    const props = {
      status: 'warning',
      text: 'TextStringWarning'
    }
    const component = renderer.create(
      <StatusField  {...props} />
    )
    expect(component.toJSON()).toMatchSnapshot()
  })
  it('renders failed as expected', () => {
    const props = {
      status: 'failed',
      text: 'TextStringFailed'
    }
    const component = renderer.create(
      <StatusField  {...props} />
    )
    expect(component.toJSON()).toMatchSnapshot()
  })
  it('renders default as expected', () => {
    const props = {
      status: 'unknown',
      text: 'TextStringDefault'
    }
    const component = renderer.create(
      <StatusField  {...props} />
    )
    expect(component.toJSON()).toMatchSnapshot()
  })
})
