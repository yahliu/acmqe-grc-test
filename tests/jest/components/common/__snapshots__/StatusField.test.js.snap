// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatusField renders default as expected 1`] = `
<span
  className="table-status-row"
>
  <span
    className="table-status-icon table-status-icon__unknown"
  >
    <svg
      aria-hidden={true}
      aria-labelledby={null}
      fill="#f0ab00"
      height="1em"
      role="img"
      style={
        Object {
          "verticalAlign": "-0.125em",
        }
      }
      viewBox="0 0 576 512"
      width="1em"
    >
      <path
        d="M569.517 440.013C587.975 472.007 564.806 512 527.94 512H48.054c-36.937 0-59.999-40.055-41.577-71.987L246.423 23.985c18.467-32.009 64.72-31.951 83.154 0l239.94 416.028zM288 354c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z"
      />
    </svg>
  </span>
  <span
    className="table-status-label"
  >
    TextStringDefault
  </span>
</span>
`;

exports[`StatusField renders failed as expected 1`] = `
<span
  className="table-status-row"
>
  <span
    className="table-status-icon table-status-icon__failed"
  >
    <svg
      aria-hidden={true}
      aria-labelledby={null}
      fill="#c9190b"
      height="1em"
      role="img"
      style={
        Object {
          "verticalAlign": "-0.125em",
        }
      }
      viewBox="0 0 512 512"
      width="1em"
    >
      <path
        d="M504 256c0 136.997-111.043 248-248 248S8 392.997 8 256C8 119.083 119.043 8 256 8s248 111.083 248 248zm-248 50c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z"
      />
    </svg>
  </span>
  <span
    className="table-status-label"
  >
    TextStringFailed
  </span>
</span>
`;

exports[`StatusField renders ok as expected 1`] = `
<span
  className="table-status-row"
>
  <span
    className="table-status-icon table-status-icon__ok"
  >
    <svg
      aria-hidden={true}
      aria-labelledby={null}
      fill="#3e8635"
      height="1em"
      role="img"
      style={
        Object {
          "verticalAlign": "-0.125em",
        }
      }
      viewBox="0 0 512 512"
      width="1em"
    >
      <path
        d="M504 256c0 136.967-111.033 248-248 248S8 392.967 8 256 119.033 8 256 8s248 111.033 248 248zM227.314 387.314l184-184c6.248-6.248 6.248-16.379 0-22.627l-22.627-22.627c-6.248-6.249-16.379-6.249-22.628 0L216 308.118l-70.059-70.059c-6.248-6.248-16.379-6.248-22.628 0l-22.627 22.627c-6.248 6.248-6.248 16.379 0 22.627l104 104c6.249 6.249 16.379 6.249 22.628.001z"
      />
    </svg>
  </span>
  <span
    className="table-status-label"
  >
    TextStringOK
  </span>
</span>
`;

exports[`StatusField renders warning as expected 1`] = `
<span
  className="table-status-row"
>
  <span
    className="table-status-icon table-status-icon__warning"
  >
    <svg
      aria-hidden={true}
      aria-labelledby={null}
      fill="#f0ab00"
      height="1em"
      role="img"
      style={
        Object {
          "verticalAlign": "-0.125em",
        }
      }
      viewBox="0 0 576 512"
      width="1em"
    >
      <path
        d="M569.517 440.013C587.975 472.007 564.806 512 527.94 512H48.054c-36.937 0-59.999-40.055-41.577-71.987L246.423 23.985c18.467-32.009 64.72-31.951 83.154 0l239.94 416.028zM288 354c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z"
      />
    </svg>
  </span>
  <span
    className="table-status-label"
  >
    TextStringWarning
  </span>
</span>
`;
